### PR TITLE
Don't make a mongodb connection required

### PR DESCRIPTION
**Pull request by suneettipirneni**

### CODE REVIEW

**Review by Armintron — COMMENTED** [review]
Shouldn't we instead not instantiate the storage service in general and just check if it's null? This checks if you specifically dont have env vars, but doesnt check if you have a valid mongo connection

**Comment by suneettipirneni** [comment]
> Shouldn't we instead not instantiate the storage service in general if you dont have the vars and just check if it's null? This checks if you specifically dont have env vars, but doesnt check if you have a valid mongo connection

If you don't have the env vars set you can't have a valid connection. MongoDB was never required until my changes to plugin state.

**Review by Armintron — COMMENTED** [review]
Isn't this logic specific to the function? Why not put it in the function?

**Review by Armintron — APPROVED** [review]

File: src/services/plugin.service.ts [hunk 1]
```diff
@@ -21,6 +21,11 @@ export class PluginService {
 
   public async initPluginState(container: IContainer): Promise<void> {
 
+    if (!process.env.MONGO_DB_NAME || !process.env.MONGO_URL ||
+      !process.env.MONGO_USER_NAME || !process.env.MONGO_USER_PASS) {
+        return;
+    }
```
Comment (tanndlin) [inline src/services/plugin.service.ts:27]: Does this default to on?

File: src/services/plugin.service.ts [hunk 2]
```diff
@@ -21,6 +21,11 @@ export class PluginService {
 
   public async initPluginState(container: IContainer): Promise<void> {
 
+    if (!process.env.MONGO_DB_NAME || !process.env.MONGO_URL ||
```
Comment (c650) [inline src/services/plugin.service.ts:24]: prefer using `Environment`, not `process.env`

File: src/services/plugin.service.ts [hunk 3]
```diff
@@ -21,6 +21,11 @@ export class PluginService {
...
+      !process.env.MONGO_USER_NAME || !process.env.MONGO_USER_PASS) {
+        return;
+    }
+
     if (!mongoose.connection.readyState) {
       await container.storageService.connectToDB();
     }
```
Comment (c650) [inline src/services/plugin.service.ts:31]: cant we just try/catch instead

File: src/services/plugin.service.ts (hunk 2 above)
Comment (suneettipirneni) [reply src/services/plugin.service.ts:24]: `Environment` was removed

Comment (c650) [reply src/services/plugin.service.ts:24]: seethe

File: src/services/plugin.service.ts (hunk 3 above)
Comment (suneettipirneni) [reply src/services/plugin.service.ts:31]: This is for connecting to the database if not already connected. If we get to this point, we should be able to connect, and no errors should be thrown.

**Review by c650 — APPROVED** [review]

**Review by CallumCoy — APPROVED** [review]
lgtm, and Thank you.

**Review by Armintron — APPROVED** [review]

**Pull request merged**